### ISSUE TITLE
Audio/video speed up

### ISSUE THREAD

**Issue by Rohirrimus** [Grippentech/Asus-E200HA-Linux-Post-Install-Script#9]
Hello,
So I follow your script and install everything as said but when I do so after I reboot when I play something (youtube video, song) there is no audio and now it goes 10 times faster. 30sec of youtube video played in just a few sec. 
Tried Kubuntu 17.4 and new Deepin OS. Same problem.
Is there any fix for this,because as of right now script only makes things worse.

(Note: Also tried new bleeding edge kernel 4.13 from Ubuntu kernel ppa,did not fix the problem.)

**Comment by Grippy98** [comment]
Hey, as of right now this is only confirmed to work properly on Ubuntu 16.10 or 17.04. Kubuntu SHOULD work but it depends on the individual packages for the distro. Please try normal Ubuntu, if it works fine, just install kubuntu-desktop as a package separately. Otherwise please post back.

Sorry to not have a better solution.

**Comment by Rohirrimus** [comment]
Alright,so I tried Ubuntu, just installed 17.4 and still no luck. Audio doesn't work after script, uname show kernel is updated and pulsefire version is 10.0.
Video plays fine now but no audio. Would maybe going up to 4.13 kernel fix it?

**Comment by Grippy98** [comment]
Hmm bizarre. Going to Kernel 4.13 wouldn't fix it since I don't believe the
changes in the version I have compiled were put in the mainline kernel yet.
What audio devices show up in the gui?

On May 22, 2017 1:11 PM, "Rohirrimus" <notifications@github.com> wrote:

Alright,so I tried Ubuntu, just installed 17.4 and still no luck. Audio
doesn't work after script, uname show kernel is updated and pulsefire
version is 10.0.
Video plays fine now but no audio. Would maybe going up to 4.13 kernel fix
it?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/Grippentech/Asus-E200HA-Linux-Post-Install-Script/issues/9#issuecomment-303162474>,
or mute the thread

**Comment by Rohirrimus** [comment]
Thank you for help,
Here you can see screen-shoots. ( I am on ubuntu now,but while I was on Kubuntu I saw it as Intel-something).

![screenshot from 2017-05-22 21-37-26](https://cloud.githubusercontent.com/assets/9285547/26325517/7cbbe0c8-3f37-11e7-9939-02fe7058052a.png)
![screenshot from 2017-05-22 21-34-18](https://cloud.githubusercontent.com/assets/9285547/26325518/7cbd3892-3f37-11e7-927b-f9c26372a4a4.png)

**Comment by Grippy98** [comment]
Looks like it's not loading the kernel correctly. Select it at boot-time or uninstall the different versions you have installed. The one my script compiles ends in -custom not -generic. 

That's why the audio device isn't recognized.

**Comment by Rohirrimus** [comment]
It worked. And then not again.
It started working, however when I installed Kubuntu-desktop, It stopped again. I made sure I am loading into the right kernel with Kubuntu desktop but no luck. 
Is there any other system that is tested to work besides Stock Ubuntu ( I hate Unity) ? Thanks

**Comment by Grippy98** [comment]
I've had reports of Ubuntu Mate working fine and I've personally used GNOME with it fine. It might just be KDE being weird with what services it installs.

**Comment by ghost** [comment]
looks like your linux is booting the wrong kernel. Cause this is the default kernel version. 
If you see the Grub Bootloader select kernel 4.10-custom. 

Realy you mean that it could reside on the DE?

**Comment by Grippy98** [comment]
Yes if you look back I did say he's booting the wrong kernel version. And no it does not reside on the desktop environment, in fact I don't think any should cause issues but a lot of different apps and services are installed along with your DE so idk, always a possibility.

**Comment by Rohirrimus** [comment]
Got it to work on KDE for a single boot.
When I booted it worked fine, and after reboot, I am getting it working half way.
on Kubuntu desktop from taskbar audio devices, there is none shown and audio is shown as the red line (muted). Audio is also not playing.
However, when I go into audio manager I do see the audio device and pressing test left/right works.

I tried adding to  /etc/pulse/default.pa
set-default-sink alsa_output.platform-cht-cx2072x.HiFi__hw_chtcx2072x_0__sink
but no luck.
Could I provide you with any info to help fix this?
Thank you

**Comment by Lucassifoni** [comment]
Hey, I'm running a X206H and I was having the same problem with this script under Lubuntu 17.04 (and no sound, no sd card, etc).
I added set-default-sink alsa_output.platform-cht-cx2072x.HiFi__hw_chtcx2072x_0__sink to pulseaudio, with no luck.

I upgraded to Lubuntu 17.10 alpha (feeling adventurous), then re-ran your script.

### Under Lubuntu 17.10alpha :
- [x] Bluetooth works
- [x] SD card reader works (no Sandisk still, but whatever, very good results with a Transcend SD card)
- [x] Sound works reliably :+1:, switches from speakers to headphones automatically
- [x] Wifi works (but I never had problems with it)
- [x] Brightness works
- [x] Battery life didn't change
- [x] My USB3 peripherals work fine

**Comment by aaguilarjimenez** [comment]
Hey Lucassifoni,
I am running Ubuntu MATE 17.04 on the Asus E200HA, and considering upgrading to the 17.10beta. So:
- Did the mic also work for you? Mine doesn't, at all.
- Which kernel did you install, the 4.12rc1 or the 4.13rc6? The latter didn't work that great for me, and reverted back to the 4.12rc1...
- What about suspension? It really suspends the system when you close the lid? Mine it does look like it, but the battery goes down way too fast for being really suspended...

**Comment by Lucassifoni** [comment]
Hi,
I'm under 4.12.0-rc1-custom3 right now. The mic doesn't work yet. Neither does suspension, since this script disables it and just shuts the screen down.
I'm sure it's a matter of weeks/months until everything works its way into mainline :) .

**Comment by Grippy98** [comment]
Yeah stick to 4.12rc1 for now guys, 4.13rc6 is super broken, working on fixing it. I'm putting a note to users to not download that yet (it's why it's in the dev branch of the script). 

I haven't played with suspension on the latest kernels since it used to work sometimes and cause the system to hang at random when waking up but I guess I should give that a try. 

I'll try to get a working 4.13rc6 build this weekend. Supposedly 4.13 has a lot of sound merges from Takashi, altho I'm not sure the particular Connexant codex we need is in there as I've seen no mention of it. 

And indeed microphone doesn't work yet. :)

**Comment by aaguilarjimenez** [comment]
Andrei, I doubt that you'll make suspension work with kernel 4.13. Takashi just committed yesterday (for 4.14) a patch reverting some changes made in 4.13 that were preventing the S0i3 suspension state:

https://git.kernel.org/pub/scm/linux/kernel/git/tiwai/sound.git/commit/drivers/clk/x86/clk-pmc-atom.c?h=topic/asus-e100h-4.13

This is the follow-up bugzilla entry created to deal with this regression:
https://bugzilla.kernel.org/show_bug.cgi?id=196861

I think we won't get suspension working until a RC based on 4.14.
Still, I can live with that for now. I'm more concerned about getting the mic to work properly. Will gladly try your next patched 4.13 kernel  :)

**Comment by Grippy98** [comment]
Yeah looks that way... it's a shame really. At this point I hope 4.14 will fix the mic issues as well. For 4.13 I just want to get a min configuration of the kernel that actually works with sound enabled. Something about doing a module list based on just what the laptop actually loads makes working kernels that fail to enable audio so that's going to be fun to troubleshoot. 

I haven't had time to keep up with any of this for one reason or another (school right now) so thanks a lot for keeping me up to date!

**Comment by kmbt** [comment]
Hi, do you have any idea on how to merge the `topic/asus-e100h-4.13` branch from Takashi with a current (4.15.1) kernel? As I understand, this branch is the best place to get support for cx2072x, however it lags a bit behind the current kernel version.
